src/task-controller.js

```diff
@@ -17,30 +17,61 @@
 import {SCHEDULER_PRIORITIES} from './scheduler-priorities.js';
 
 /**
- * Makes the controller's signal a TaskSignal by adding a read-only priority
- * property.
+ * The TaskSignal interface represents a signal object that allows you to
+ * communicate with a prioritized task, and abort it or change the priority
+ * via a TaskController object.
+ */
+class TaskSignal extends AbortSignal {
+  /**
+   * The priority of the task, user-visible by default.
+   * @readonly
+   * @type {string}
+   */
+  get priority() {
+    return 'user-visible';
+  }
+
+  /**
+   * The callback to be called when the priority of the task changes.
+   * @param {Function} callback
+   */
+  set onprioritychange(callback) {
+    if (this.onprioritychange_) {
+      this.removeEventListener('prioritychange', this.onprioritychange_);
+    }
+    this.addEventListener('prioritychange', callback);
+    /**
+     * @private
+     * @type {Function}
+     */
+    this.onprioritychange_ = callback;
+  }
+
+  /**
+   * The callback to be called when the priority of the task changes.
+   * @type {Function}
+   */
+  get onprioritychange() {
+    return this.onprioritychange_ || null;
+  }
+}
+
+/**
+ * Makes the TaskSignal instance from the AbortController instance.
  * @private
  * @param {TaskController} controller
  */
 function makeTaskSignal(controller) {
-  const signal = controller.signal;
-  Object.defineProperties(signal, {
-    priority: {
-      get: function() {
-        return controller.priority_;
-      },
-      enumerable: true,
-    },
-    onprioritychange: {
-      value: null,
-      writable: true,
-      enumerable: true,
+  // Change the prototype of the signal to TaskSignal.prototype.
```

Comment:
nit: You can delete this comment and the one below since it's obvious from the code itself.

Reply:
[Remove redundant comment](https://github.com/GoogleChromeLabs/scheduler-polyfill/pull/19/commits/08a33c0313aef19f489742c3252c3118dd94390f)